test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/ExtensionsTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Functions.Worker.Sdk.Generators;
+using Xunit;
+
+namespace Microsoft.Azure.Functions.SdkGeneratorTests
+{
+    public partial class FunctionMetadataProviderGeneratorTests
+    {
+        public class ExtensionsTests
+        {
+            [Theory]
+            [InlineData("HttpTriggerAttribute", "Attribute", "HttpTrigger")]
+            [InlineData("QueueOutput", "Output", "Queue")]
+            [InlineData("QueueInput", "Input", "Queue")]
```

Comment:
Consider adding an entry where the toTrim value is not a substring of input.

`[InlineData("Foo", "Input", "Foo")]`